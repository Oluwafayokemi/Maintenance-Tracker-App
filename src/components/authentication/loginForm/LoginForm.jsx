import React from 'react';

const LoginForm = ({handleChange, handleSubmit, value}) => (
  <React.Fragment>
    <div className="col-6 flex-container">
      <div className="nav">
        <nav>
          <ul className="rform">
            <form id="signin-form" className="one" onSubmit={handleSubmit}>
              <li className="hide2">Sign In</li>
              <li>
                <div className="email">
                  <label htmlFor="email">
                    <strong>Email</strong>
                  </label>
                  <input
                    type="email"
                    placeholder="Enter Email"
                    name="email"
                    value={value.email}
                    onChange={handleChange}
                    required
                  />
                </div>
              </li>
              <li>
                <div className="psw">
                  <label htmlFor="password">
                    <strong>Password</strong>
                  </label>
                  <input
                    type="password"
                    placeholder="Enter Password"
                    name="password"
                    value={value.password}
                    onChange={handleChange}
                    required
                  />
                </div>
              </li>
              <button type="submit" value="submit" id="login-btn">
                Login
              </button>
            </form>
          </ul>
        </nav>
      </div>
    </div>
  </React.Fragment>
);

export default LoginForm;
