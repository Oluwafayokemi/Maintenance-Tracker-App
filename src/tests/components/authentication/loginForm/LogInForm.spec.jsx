import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../../../../components/authentication/loginForm/LoginForm';

const mockEvent = { target: {}, preventDefault: () => jest.fn() };

describe('Tests SignInForm component', () => {
  const props = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    value: {
      email: 'email',
      password: 'password',
    },
  };
  it('should render without errors', (done) => {
    const wrapper = shallow(<LoginForm {...props} />);
    wrapper.find('[name="email"]').simulate('change', { target: { value: 'test@email.com' } });
    wrapper.find('[name="password"]').simulate('change', { target: { value: 'Password' } });
    wrapper.find('form').simulate('submit', mockEvent);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
