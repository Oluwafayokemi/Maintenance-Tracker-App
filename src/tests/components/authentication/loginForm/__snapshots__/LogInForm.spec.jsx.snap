// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests SignInForm component should render without errors 1`] = `
<Fragment>
  <div
    className="col-6 flex-container"
  >
    <div
      className="nav"
    >
      <nav>
        <ul
          className="rform"
        >
          <form
            className="one"
            id="signin-form"
            onSubmit={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "preventDefault": [Function],
                      "target": Object {},
                    },
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          >
            <li
              className="hide2"
            >
              Sign In
            </li>
            <li>
              <div
                className="email"
              >
                <label
                  htmlFor="email"
                >
                  <strong>
                    Email
                  </strong>
                </label>
                <input
                  name="email"
                  onChange={
                    [MockFunction] {
                      "calls": Array [
                        Array [
                          Object {
                            "target": Object {
                              "value": "test@email.com",
                            },
                          },
                        ],
                        Array [
                          Object {
                            "target": Object {
                              "value": "Password",
                            },
                          },
                        ],
                      ],
                      "results": Array [
                        Object {
                          "isThrow": false,
                          "value": undefined,
                        },
                        Object {
                          "isThrow": false,
                          "value": undefined,
                        },
                      ],
                    }
                  }
                  placeholder="Enter Email"
                  required={true}
                  type="email"
                  value="email"
                />
              </div>
            </li>
            <li>
              <div
                className="psw"
              >
                <label
                  htmlFor="password"
                >
                  <strong>
                    Password
                  </strong>
                </label>
                <input
                  name="password"
                  onChange={
                    [MockFunction] {
                      "calls": Array [
                        Array [
                          Object {
                            "target": Object {
                              "value": "test@email.com",
                            },
                          },
                        ],
                        Array [
                          Object {
                            "target": Object {
                              "value": "Password",
                            },
                          },
                        ],
                      ],
                      "results": Array [
                        Object {
                          "isThrow": false,
                          "value": undefined,
                        },
                        Object {
                          "isThrow": false,
                          "value": undefined,
                        },
                      ],
                    }
                  }
                  placeholder="Enter Password"
                  required={true}
                  type="password"
                  value="password"
                />
              </div>
            </li>
            <button
              id="login-btn"
              type="submit"
              value="submit"
            >
              Login
            </button>
          </form>
        </ul>
      </nav>
    </div>
  </div>
</Fragment>
`;
